filesystem_service/src/filesystem_service/monitor.py

```diff
@@ -0,0 +1,169 @@
+"""
+Provides functionality to monitor local filesystem changes
+"""
+
+from multiprocessing import Process
+import os
+
+import inotify.adapters
+
+
+class FileSystemMonitor(object):
+    """
+    Monitor local filesystem changes and inform subscribers
+    """
+
+    def __init__(self):
+        self._folder = None
+        self._exclude_folder = set()
+        self._process = None
+        self._subscribers = set()
+        self.notifier = inotify.adapters.Inotify()
+
+    @property
+    def subscribers(self):
+        """
+        :return: List of subscribers
+        """
+
+        return self._subscribers
+
+    def register_subscriber(self, subs):
+        """
+        Register new subscriber
+        :param subs: an instance of subscriber
+        :return:
+        """
+
+        self._subscribers.add(subs)
+
+    def unregister_subscriber(self, subs):
+        """
+        Remove subscriber
+        :param subs: an instance of subscriber
+        :return:
+        """
+
+        self._subscribers.remove(subs)
+
+    @property
+    def exclude_folder(self):
+        """
+        :return: list of folders to be excluded from monitoring
+        """
+
+        return self._exclude_folder
+
+    def add_exclude_folder(self, folder):
+        """
+        :param folder:full path to folder. Type: string
+        :return:
+        """
+
+        self._exclude_folder.add(folder.encode())
+
+    @property
+    def folder(self):
+        """
+        :return: Monitored folder
+        """
+
+        return self._folder
+
+    @folder.setter
+    def folder(self, folder):
+        """
+        :param folder: full path to folder to be monitored. Type: string
+        :return:
+        """
+
+        self._folder = folder.encode()
+
+    def start(self):
+        """
+        Start process that generates filesystem events
+        :return:
+        """
+
+        self._process = Process(target=self.notify)
```

Comment:
It is not supposed to spawn a new process, but to start monitoring in the same thread.

Reply:
done